src/Server/MySQLHandler.cpp

```diff
@@ -544,6 +548,18 @@ static bool isFederatedServerSetupSetCommand(const String & query)
     return re2::RE2::FullMatch(query, expr);
 }
 
+/// Always return an empty set with appropriate column definitions for SHOW WARNINGS queries
```

Comment:
Thoughts:
- as the implementation is in MySQLHandler.cpp, I assume the statements work only through the MySQL port?
- similarity, I was thinking whether the statements should be documented in docs/en/sql-reference/statements/show.md but maybe that's not necessary when exist only for the MySQL protocol
- we are doing simple string comparisons (l. 91/92) to handle the statements. While matching is case-insensitive (good), statements with more than one whitespace (`SHOW     WARNINGS`) are to the user's surprise not recognized but I guess this isn't terrible for our use case. Maybe you like to add a comment about that (l. 90 maybe?)

Reply:
p.1 - yes, it works only via MySQL port.
p.2 - docs are probably not necessary, indeed.
p.3 is precisely why I am not very happy with the current implementation; it is fine for automatically generated statements (via tools), but it falls apart with any typos, like extra spaces. I'd say when we accumulate all the current fixes and related tests (including the rolled back #58835) AND if we have issues with queries matching (extra spaces, etc), we can think about a more sophisticated approach. 

The extra spaces issue, in particular, is true for any query that was historically hardcoded into the MySQLHandler (I just added new ones in a similar fashion).